test/services/dangling_account_remover_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class DanglingAccountRemoverTest < ActiveSupport::TestCase
+  test 'removes accounts that have no relationships' do
+    FactoryBot.create_list(:account, 30)
+
+    assert_difference('Account.count' => -30) do
+      DanglingAccountRemover.run!
+    end
+  end
+
+  test 'keeps accounts that have hosts with test results or policies' do
```

Comment:
Variants I expected in prod:

- keeps accounts that have policies but no hosts
- keeps accounts that have profiles but no policies or hosts
- removes users along with removed accounts